scc/src/opts.rs

```diff
@@ -50,85 +50,189 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
     pub cache_file: Option<PathBuf>,
 }
 
-fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<bool> {
+impl Default for Opts {
+    fn default() -> Self {
+        Self {
+            script_paths: vec![],
+            cache_dir: None,
+            optimize: false,
+            threads: 1,
+            warnings: None,
+            no_testonly: false,
+            no_breakpoint: false,
+            profile: false,
+            script_paths_file: None,
+            cache_file: None,
+        }
+    }
+}
+
+fn no_space(name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
     any::<String>(name)
         .help(help)
-        .guard(|s| s.starts_with('-'), "doesn't start with -")
+        .guard(
+            |s| s.starts_with(name.to_owned().as_str()),
+            "no_space doesn't start with -<name>",
+        )
         .parse(move |s| {
-            let Some(cur_name) = s.strip_prefix('-') else {
-                return Err(format!("{} is not a toggle option", s));
-            };
-            if cur_name != name {
-                Err(format!("{} is not a known toggle option name", cur_name))
+            // println!("no_space: {}", s);
+            s.strip_prefix(name.to_owned().as_str())
+                .ok_or(Err::<String, &str>("IDK1"))
+                .map_or(Err("IDK2"), |s| Ok(s.to_string()))
+        })
+        .anywhere()
+        .optional()
+        .catch()
+}
+
+fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<Option<bool>> {
+    any::<String>(name)
+        .help(help)
+        .guard(|s| s.eq(name.to_owned().as_str()), "toggle_option isn't -<name>")
+        .parse(move |s| {
+            // println!("toggle_options: {}", s);
+            if s != name {
+                Err(format!("{} is not a known toggle option name", s))
             } else {
                 Ok(true)
             }
         })
+        .adjacent()
         .anywhere()
-        .fallback(false)
+        .optional()
+        .catch()
 }
 
-fn slong(tag_str: &'static str, arg_str: &'static str) -> impl Parser<String> {
-    let tag = any::<String>(tag_str).guard(move |s| s.as_str() == tag_str, "not arg name");
-    let value = positional::<String>(arg_str);
-    construct!(tag, value).anywhere().map(|pair| pair.1)
+fn equals_sign(name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
+    any::<String>(name)
+        .help(help)
+        .guard(
+            |s| s.starts_with(format!("{}=", name.to_owned()).as_str()),
+            "equals_sign doesn't start with -<name>=",
+        )
+        .parse(move |s| {
+            // println!("equals_sign: {}", s);
+            let arg = s.split('=').collect::<Vec<_>>();
+            let cur_name = arg.first().unwrap().to_owned();
+            if cur_name != name {
+                Err(format!("{} is not a known option name", cur_name))
+            } else {
+                Ok((*arg.last().unwrap()).to_string())
+            }
```

Comment:
you should not use `unwrap` here because it'll cause a panic in the parser and fail to propagate the error
also, `split_once` can be used to simplify the code:
```suggestion
            let (_, val) = s.split_once('=').ok_or("actual error message")?;
            Ok::<_, &str>(val.to_owned())
```